File: civic_voice/replace_opacity.py
import os
import re

lib_dir = "c:\\CIVIC VOICE AI FOR BHARAT\\civic_voice\\lib"

for root, dirs, files in os.walk(lib_dir):
    for file in files:
        if file.endswith(".dart"):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Use regex to find .withOpacity(value) and replace with .withValues(alpha: value)
            new_content = re.sub(r'\.withOpacity\(([^)]+)\)', r'.withValues(alpha: \1)', content)
            
            if new_content != content:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Updated {path}")

File: civic_voice/replace_opacity.dart
import 'dart:io';

void main() {
  final dir = Directory('lib');
  int count = 0;
  
  for (final file in dir.listSync(recursive: true)) {
    if (file is File && file.path.endsWith('.dart')) {
      var content = file.readAsStringSync();
      
      final regex = RegExp(r'\.withOpacity\(([^)]+)\)');
      final newContent = content.replaceAllMapped(regex, (match) {
        return '.withValues(alpha: ${match.group(1)})';
      });
      
      if (content != newContent) {
        file.writeAsStringSync(newContent);
        final matches = regex.allMatches(content).length;
        count += matches;
        print('Updated ${file.path} ($matches replacements)');
      }
    }
  }
  
  print('Total replacements: $count');
}
